cf-oscore/src/main/java/org/eclipse/californium/oscore/OSCoreCtx.java

```diff
@@ -689,14 +689,22 @@ public void setSeqMax(int seqMax) {
 	 */
 	private void setLengths() {
 		if (common_alg != null) {
-			if (common_alg.equals(AlgorithmID.AES_CCM_16_64_128)) {
+
+			switch (common_alg) {
+			case AES_CCM_16_64_128:
+			case AES_CCM_64_64_128:
+			case AES_CCM_16_128_128:
+			case AES_CCM_64_128_128:
 				iv_length = ivLength(common_alg);
-				id_length = 7;
-				key_length = common_alg.getKeySize() / 8; // 16;
-			} else {
+				id_length = iv_length - 6; // RFC section 5.2
```

Comment:
If the iv-length is 7, the resulting id-length is 1. Is that intended?

Reply:
Yes, that is correct. That is why OSCORE only supports AEAD algorithms with nonces 7 bytes or more.

Having a maximum length of the sender/recipient ID be 1 byte only is small, but there is also the ID Context that can be used to identify a specific context (if more than 256 are needed).